lib/srv/app/server.go

```diff
@@ -228,12 +242,15 @@ func New(ctx context.Context, c *Config) (*Server, error) {
 		dynamicLabels: make(map[string]*labels.Dynamic),
 		apps:          make(map[string]types.Application),
 		awsSigner:     awsSigner,
+		connLimiter:   c.ConnectionLimiter.ConnectionsLimiter,
 		monitoredApps: monitoredApps{
 			static: c.Apps,
 		},
 		reconcileCh: make(chan struct{}),
 	}
 
+	c.ConnectionLimiter.WrapHandle(s)
```

Comment:
What do you need WrapHandle here for if you're doing Acquire/Release explicitly below?